blob/driver/driver.go

```diff
@@ -107,6 +107,44 @@ type Attributes struct {
 	AsFunc func(interface{}) bool
 }
 
+// ListOptions sets options for listing objects in the bucket.
+// TODO(Issue #541): Add Delimiter.
+type ListOptions struct {
+	// Prefix indicates that only results with the given prefix should be
+	// returned.
+	Prefix string
+
+	// PageSize sets the maximum number of objects that will be returned in
+	// a single call. It is guaranteed to be > 0 and <= blob.MaxPageSize.
+	PageSize int
+	// PageToken may be filled in with the NextPageToken from a previous
+	// ListPaged call.
+	PageToken []byte
+}
+
+// ListObject represents a specific blob object returned from ListPaged.
+type ListObject struct {
+	// Key is the key for this blob.
+	Key string
+	// ModTime is the time the blob object was last modified.
+	ModTime time.Time
+	// Size is the size of the object in bytes.
+	Size int64
+}
+
+// ListPage represents a page of results return from ListPaged.
+type ListPage struct {
+	// Objects is the slice of objects found. It should have at most
+	// ListOptions.PageSize entries.
+	Objects []*ListObject
+	// NextPageToken should be left empty unless
+	// len(Objects) == ListOptions.PageSize and there are more objects to
```

Comment:
This comment needs to be fixed then: it should be permissible for the driver to return a smaller page than requested.

Reply:
Will do.